fedbiomed/researcher/experiment.py

```diff
@@ -1643,6 +1647,36 @@ def run_once(self, increase: bool = False, test_after: bool = False) -> int:
 
         return 1
 
+    def _process_optim_aux_var(
+        self,
+    ) -> None:
+        """Process any Optimizer auxiliary variables received during a round.
+
+        Raises:
+            FedbiomedExperimentError: if auxiliary variables were received,
+                but `agg_optimizer` is None and thus cannot process them.
+            FedbiomedOptimizerError: if the received auxiliary variables do
+                not match the expectations of the `agg_optimizer` Optimizer.
+        """
+        # Restructure the received auxiliary variables (if any).
+        aux_var = {}  # type: Dict[str, Dict[str, Dict[str, Any]]]
+        for reply in self._job.training_replies[self._round_current]:
+            node_id = reply["node_id"]
+            aux_var = reply.get("optim_aux_var", {})
+            for module, params in aux_var.items():
+                aux_var.setdefault(module, {})[node_id] = params
+        # If an Optimizer is used, pass it the auxiliary variables (if any).
+        if self._agg_optimizer is not None:
+            self._agg_optimizer.set_aux(aux_var)
+        # If no Optimizer is used but auxiliary variables were received, raise.
```

Comment:
declearn already handles that for Vector-encapsulated data, based on GPU availability and the global device policy - which we should handle in Fed-BioMed (using `declearn.utils.set_device_policy` in the parts that already take care of selecting whether to use CPU or GPU).

Reply:
OK then I guess we may want to open a separate issue on that, I feel it can be interesting to have but doesnot belong to this Pull Request

Reply:
I agree - and I think that this issue (dealing with device placement when using declearn) should be our next priority after the current MR. Note that I also think that it will be straightforward to address thanks to `declearn.utils.set_device_policy`.